scripts/check-license/src/main.rs

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: Apache-2.0, MIT
+use regex::Regex;
+use std::collections::HashSet;
+use std::fs::{self, OpenOptions};
+use std::io::{self, BufRead, BufReader, Write};
+use std::path::{Path, PathBuf};
+use walkdir::WalkDir;
+
+const HEADER_LINES_TO_CHECK: usize = 3;
+
+lazy_static::lazy_static! {
+    static ref PL_LICENSE: Regex = Regex::new(r"// Copyright \\d{4}-\\d{4} Protocol Labs").unwrap();
+    static ref CS_LICENSE: Regex = Regex::new(r"// Copyright \\d{4}-\\d{4} ChainSafe Systems").unwrap();
+    static ref SPDX_LICENSE: Regex = Regex::new(r"// SPDX-License-Identifier: Apache-2.0, MIT").unwrap();
+
+    /// LICENSE_CHECKS is a static vector containing tuples that define license validation rules.
+    /// Each tuple consists of:
+    ///  - a regular expression to detect a license pattern,
+    ///  - a boolean indicating whether the license is mandatory, and
+    ///  - the text to add to file if license is not found.
+    static ref LICENSE_CHECKS: Vec<(&'static Regex, bool, &'static str)> = vec![
+        (&PL_LICENSE, false, "// Copyright 2021-2024 Protocol Labs"),
+        (&CS_LICENSE, false, "// Copyright 2019-2024 ChainSafe Systems"),
+        (&SPDX_LICENSE, true, "// SPDX-License-Identifier: Apache-2.0, MIT"),
+    ];
+
+    /// IGNORED_DIRECTORIES is a static set containing directories to ignore during the license check.
+    static ref IGNORED_DIRECTORIES: HashSet<PathBuf> = {
+        let mut set = HashSet::new();
+        set.insert(PathBuf::from("./target/"));
```

Comment:
```suggestion
        set.insert("./target/".into());
```

Rust is all about the magic (when it works).

Reply:
Also, it looks like this is being used as a vec.